crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -559,34 +558,52 @@ impl ReadOnlyDevice {
         Ok(())
     }
 
-    pub(crate) fn is_signed_by_device(&self, json: Value) -> Result<(), SignatureError> {
+    pub(crate) fn as_device_keys(&self) -> &DeviceKeys {
+        &self.inner
+    }
+
+    /// Check if the given JSON is signed by this device key.
+    ///
+    /// This method should only be used if a signature of an object should be
+    /// checked multiple times and the canonicalization step wants to be done
+    /// only a single time.
```

Comment:
```suggestion
    /// This method should only be used if an object's signature needs to be
    /// checked multiple times, and you'd like to avoid performing the
    /// canonicalization step each time.
```